api_app/api/routes/costs.py

```diff
@@ -29,27 +32,38 @@ def __init__(
         self.granularity = granularity
 
 
+def check_time_period(from_date: date, to_date: date):
+    if (to_date - from_date).days > 365:
```

Comment:
Should we use year?!

Reply:
the underlying service (Azure REST API) limits for one year...

Reply:
```
from dateutil.relativedelta import relativedelta
difference_in_years = relativedelta(to_date, from_date).years
```

I think it would be better to use it this way since you don't know if it was 364 in a year or 365...
doesn't have to be this way exactly but something that calculates the years differences

Reply:
good idea @LizaShak, I'll change it. thanks!